src/plugins/index_pattern_field_editor/kibana.json

```diff
@@ -3,7 +3,7 @@
   "version": "kibana",
   "server": true,
   "ui": true,
-  "requiredPlugins": ["data"],
+  "requiredPlugins": ["data", "expressions"],
```

Comment:
Interesting, why does the editor need to rely on the expression plugin?

Reply:
We import the `SerializedFieldFormat` type (https://github.com/elastic/kibana/pull/110226/files#diff-ca01e68f77305ada09a76546777dd200cb2883f531669a40cad5903c9c1dbef8R23)

Reply:
Should it be using the type from the field formats plugin instead?

@mattkime - looks like there are other places in the data plugin where a SerializedFieldFormat type is being imported from expressions and not the field_formats plugin? cc @Dosant

Reply:
It should use the fieldFormats plugin version.

I don't think importing types (and types alone) requires adding the plugin as a requiredPlugin or a requiredBundle, just adding it to the typescript config. That said, its hard for me to think of a case where that would make sense.

Reply:
I am using the same interface as used in the index pattern `data` plugin though. Don't you think it should match?

https://github.com/elastic/kibana/pull/110226/files#diff-64a353672cfc941c11c93ce19983c50f5de7ae626fa93b35a402f5588ac11c71R30

Reply:
@mattkime 
https://github.com/elastic/kibana/blob/master/src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts#L23

https://github.com/elastic/kibana/blob/master/src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts#L480

Reply:
Can we remove the dependency on "expressions" and create a follow up issue that uses the fieldFormatters copy of the type instead?

Reply:
As you seem to be more aware of the type you'd like to use do you mind changing the dependency creating a PR to the feature branch?
I am not sure which plugin you refer to and don't want to get it wrong and delay merging this work. Cheers 👍

Reply:
> looks like there are other places in the data plugin where a SerializedFieldFormat type is being imported from expressions and not the field_formats plugin

Created to clean this up: https://github.com/elastic/kibana/issues/111509


@sebelga, you could just change imports to import that type from: https://github.com/elastic/kibana/blob/77e25bedfb16dbd91914d53b961c69b4e2f0f18b/src/plugins/field_formats/common/types.ts#L119-L122 and remove `expression` dependency. 

> I don't think importing types (and types alone) requires adding the plugin as a requiredPlugin or a requiredBundle, 

As Matt mentioned, I am not sure from the top of my head if the dependency is even needed. I'd try to remove it all together first before adding `fieldFormats` dep.

But even if have to add a `fieldFormats` dep, it seems to be less confusing than `expressions` dependency here.

Reply:
Ok I will do that. I did add the dependency otherwise CI was failing. So it seems that is is required, even for types.